src/mono/mono/mini/mini-wasm-debugger.c

```diff
@@ -497,6 +500,15 @@ assembly_loaded (MonoProfiler *prof, MonoAssembly *assembly)
 	DEBUG_PRINTF (2, "assembly_loaded callback called for %s\n", assembly->aname.name);
 	MonoImage *assembly_image = assembly->image;
 	MonoImage *pdb_image = NULL;
+
+	if (!mono_is_debugger_attached ()) {
+		has_pending_lazy_loaded_assemblies = TRUE;
+		return;
+	}
+
+	if (mono_wasm_assembly_already_added(assembly->aname.name))
```

Comment:
Won't this always return true even if an assembly hasn't been loaded into the DebugStore via the event? For example.

1. User launches an app without attaching to a debugger.
2. User navigates to a page that lazily-loads an assembly.
3. Assembly gets loaded and is added to the [assemblies list](https://github.com/dotnet/runtime/blob/ddc83ad2b6f0127f81486301185c57f8e9aac055/src/mono/wasm/runtime/driver.c#L185).
4. `mono_wasm_asm_loaded` event isn't fired since debugger isn't attached.
5. User attaches to a debugger.
6. `assembly_loaded` callback is invoked from `mono_wasm_set_is_debugger_attached`.
7. This check succeeds because the assembly was added in step 3.

Let me know if my understanding of this is incorrect.

Reply:
Lazy loaded assemblies will be in the files that are passed to _handle_loaded_asset?

Reply:
They are added via `Assembly.Load` in [this code](https://github.com/dotnet/aspnetcore/blob/master/src/Components/WebAssembly/WebAssembly/src/Services/LazyAssemblyLoader.cs#L113-L119). `_load_assets_and_runtime` is only used for assemblies that are loaded when the app is initialized.

Reply:
But we only add things to [assemblies list](https://github.com/dotnet/runtime/blob/ddc83ad2b6f0127f81486301185c57f8e9aac055/src/mono/wasm/runtime/driver.c#L185), if the come from _load_assets_and_runtime, am I wrong?

Reply:
Nope! I had a gap in my understanding. This check will work fine. Resolving this comment now.